tools/src/main/java/org/apache/kafka/trogdor/fault/Kibosh.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.trogdor.fault;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.apache.kafka.trogdor.common.JsonUtil;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.TreeMap;
+
+public final class Kibosh {
+    public static final Kibosh INSTANCE = new Kibosh();
+
+    public final static String KIBOSH_CONTROL = "kibosh_control";
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME,
+        include = JsonTypeInfo.As.PROPERTY,
+        property = "type")
+    @JsonSubTypes({
+            @JsonSubTypes.Type(value = KiboshFilesUnreadableFaultSpec.class, name = "unreadable"),
+        })
+    public static abstract class KiboshFaultSpec {
+        @Override
+        public final boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            return Objects.equals(toString(), o.toString());
+        }
+
+        @Override
+        public final int hashCode() {
+            return toString().hashCode();
+        }
+
+        @Override
+        public final String toString() {
+            return JsonUtil.toJsonString(this);
+        }
+    }
+
+    public static class KiboshFilesUnreadableFaultSpec extends KiboshFaultSpec {
+        private final String prefix;
+        private final int errorCode;
+
+        @JsonCreator
+        public KiboshFilesUnreadableFaultSpec(@JsonProperty("prefix") String prefix,
+                                              @JsonProperty("errorCode") int errorCode) {
+            this.prefix = prefix;
+            this.errorCode = errorCode;
+        }
+
+        @JsonProperty
+        public String prefix() {
+            return prefix;
+        }
+
+        @JsonProperty
+        public int errorCode() {
```

Comment:
As I can see here https://github.com/confluentinc/kibosh/blob/b4288fa98d637f3765dc2abe848d18fce5afa897/fault.c#L41
the parameter name should be 'code', not 'errorCode'. Do you this fault actually working?